toffy/streak_detection_test.py

```diff
@@ -0,0 +1,178 @@
+            ),
+            corrected_channels=None,
+            streak_data=None,
+        )
+        for fov_name in fovs
+    ]
+    request.cls.fovs = fovs_dc
+    request.cls.data_dir = data_dir
+
+
+@pytest.mark.skip(reason="Cannot test with real data, need to utilize synthetic data")
+@pytest.mark.usefixtures("get_fovs")
+class TestStreakDetection:
+    def test_streak_detection(self):
+        # Test to make sure all the fovs are loaded correctly
+        assert len(self.fovs) == 3
+
+        # Test the end-to-end streak correction pipeline function
+        #   `streak_detection::streak_correction`
+        # Make sure the input shape is equal to the output shape
+        # Add the corrected_channels and the streak_data to the Fov dataclass
+        for idx, fov in enumerate(self.fovs):
+            corrected_channels, streak_data = sd.streak_correction(
+                fov_data=fov.fov_da, streak_channel="Noodle", visualization_masks=True
+            )
+            assert corrected_channels.shape == fov.fov_da[0, ...].shape
+            self.fovs[idx].corrected_channels = corrected_channels
+            self.fovs[idx].streak_data = streak_data
+
+    def test_save_corrected_channels(self):
+        with tempfile.TemporaryDirectory() as save_dir:
+
+            save_dir_path = Path(save_dir)
+
+            for fov in self.fovs:
+                # Save Streak Masks
+                sd.save_corrected_channels(
+                    streak_data=fov.streak_data,
+                    corrected_channels=fov.corrected_channels,
+                    data_dir=save_dir,
+                    save_streak_data=True,
+                )
+            # Gather all the saved files
+            fovs = [fov.stem for fov in save_dir_path.glob("*//") if "corrected" in fov.stem]
+
+            # Check that the names in the corrected files are the same as the original files
+
+            for corrected_fov, fov in zip(fovs, self.fovs):
+                corrected_tiff_names = [
+                    cor_tiff.stem
+                    for cor_tiff in Path(save_dir_path / corrected_fov).glob("*.tiff")
+                ]
+
+                # Get initial file names
+                initial_tiff_names = [
+                    init_tiff.stem
+                    for init_tiff in Path(self.data_dir / fov.fov_name).glob("*.tiff")
+                ]
+
+                # Test that the set of initial tiffs map 1-to-1 to set of corrected tiffs
+                assert Counter(initial_tiff_names) == Counter(corrected_tiff_names)
+
+
+# ? Construct different sets of fake data (noisy, and clean variants)
+# ? Modify to work with `parameterize_with_cases`
+# TODO: All channels of fake data must contain the same streaks. Add noise to each channel.
+# TODO: Add multiple streaks.
+# TODO: Hardcode the streak tests first, request for review
+@pytest.fixture(scope="class", params=[2, 4, 10])
+def fake_data(request):
```

Comment:
Rater than a single large fixture that gets passed to every function, we want to the create the simplest possible test data for each function